app/actions/websocket/channel.ts

```diff
@@ -18,42 +27,201 @@ import {getCurrentUser, getUserById} from '@queries/servers/user';
-            const {users} = await fetchUsersByIds(serverUrl, [userId], true);
-            if (users) {
-                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+        const models: Model[] = [];
+        const {channels, memberships} = await fetchMyChannel(serverUrl, teamId, channelId, true);
+        if (channels && memberships) {
+            const prepare = await prepareMyChannelsForTeam(operator, teamId, channels, memberships);
+            if (prepare) {
+                const prepareModels = await Promise.all(prepare);
+                const flattenedModels = prepareModels.flat();
+                if (flattenedModels?.length > 0) {
+                    models.push(...flattenedModels);
+                }
+                const categoryModels = await addChannelToDefaultCategory(serverUrl, channels[0], true);
+                if (categoryModels.models?.length) {
+                    models.push(...categoryModels.models);
+                }
             }
         }
+        operator.batchRecords(models);
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelUnarchiveEvent(serverUrl: string, msg: any) {
+    try {
+        await setChannelDeleteAt(serverUrl, msg.data.channel_id, 0);
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelConvertedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return;
+    }
+
+    try {
+        const channelId = msg.data.channel_id;
+        const {channel} = await fetchChannelById(serverUrl, channelId);
+        if (channel) {
+            operator.handleChannel({channels: [channel], prepareRecordsOnly: false});
+        }
+    } catch {
+        // do nothing
+    }
+}
+
+export async function handleChannelUpdatedEvent(serverUrl: string, msg: any) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return;
+    }
+
+    const updatedChannel = JSON.parse(msg.data.channel);
+    try {
+        const models: Model[] = await operator.handleChannel({channels: [updatedChannel], prepareRecordsOnly: true});
+        const infoModel = await updateChannelInfoFromChannel(serverUrl, updatedChannel, true);
+        if (infoModel.model) {
+            models.push(infoModel.model);
+        }
+        operator.batchRecords(models);
+    } catch {
+        // Do nothing
+    }
+}
+
+export async function handleChannelViewedEvent(serverUrl: string, msg: any) {
+    try {
+        const {channel_id: channelId} = msg.data;
+
+        await markChannelAsViewed(serverUrl, channelId, false);
+    } catch {
+        // do nothing
+    }
+}
+
+// This event is triggered by changes in the notify props or in the roles.
+export async function handleChannelMemberUpdatedEvent(serverUrl: string, msg: any) {
```

Comment:
does this function triggers only for the currentUser ? I guess the answer is yes but just checking

Reply:
According to the server code, this is only sent to the user receiving the change.